backend/letter/models.py

```diff
@@ -94,76 +150,96 @@ def __str__(self):
+        return (
+            f"{self.agent.name} is sender of {self.letter.name} ({self.letter.source})"
+        )
 
 
-class LetterMaterial(Field, models.Model):
-    class Surface(models.TextChoices):
-        PARCHMENT = "parchment", "parchment"
-        PAPYRUS = "papyrus", "papyrus"
-        OTHER = "other", "other"
-        UNKNOWN = "unknown", "unknown"
+class LetterDescriptionAddressee(DescriptionField, models.Model):
+    """
+    Description of a person as the addressee of a letter
+    """
 
-    surface = models.CharField(
-        choices=Surface.choices,
-        null=False,
-        blank=False,
+    letter = models.ForeignKey(
+        to=LetterDescription,
+        on_delete=models.CASCADE,
     )
-    letter = models.OneToOneField(
-        to=Letter,
+    agent = models.ForeignKey(
+        to=AgentDescription,
         on_delete=models.CASCADE,
-        null=False,
     )
 
+    def clean(self):
+        if self.letter.source != self.agent.source:
+            raise ValidationError("Can only link descriptions in the same source text")
+
     def __str__(self):
-        if self.letter:
-            return f"material of {self.letter}"
-        else:
-            return f"material #{self.id}"
-
-
-# class LetterSenders(Field, models.Model):
-#     senders = models.ManyToManyField(
-#         to=Agent,
-#         blank=True,
-#         help_text="Agents whom the letter names as the sender",
-#     )
-#     letter = models.OneToOneField(
-#         to=Letter,
-#         on_delete=models.CASCADE,
-#         null=False,
-#     )
-
-#     def __str__(self):
-#         if self.letter:
-#             return f"senders of {self.letter}"
-#         else:
-#             return f"senders #{self.id}"
-
-
-# class LetterAddressees(Field, models.Model):
-#     addressees = models.ManyToManyField(
-#         to=Agent,
-#         blank=True,
-#         help_text="Agents whom the letter names as the addressee",
-#     )
-#     letter = models.OneToOneField(
-#         to=Letter,
-#         on_delete=models.CASCADE,
-#         null=False,
-#     )
-
-#     def __str__(self):
-#         if self.letter:
-#             return f"addressees of {self.letter}"
-#         else:
-#             return f"addressees #{self.id}"
+        return f"{self.agent.name} is addressee of {self.letter.name} ({self.letter.source})"
+
+
+class PreservedLetter(HistoricalEntity, models.Model):
+    """
+    A letter that has been preserved
+    """
+
+    persons_involved = models.ManyToManyField(
+        to=HistoricalPerson,
+        through="PreservedLetterRole",
+        blank=True,
+        help_text="historical figures related to the letter",
+    )
+
+
+class PreservedLetterRole(Field, models.Model):
+    """
+    Relationship between a preserved letter and a historical figure
+    """
+
+    letter = models.ForeignKey(
+        to=PreservedLetter,
+        on_delete=models.CASCADE,
+    )
+    person = models.ForeignKey(
```

Comment:
So for the moment we are not recording what the nature of the person/agent's involvement in this letter is?